community/kernel/src/main/java/org/neo4j/kernel/impl/index/schema/config/SpatialIndexSettings.java

```diff
@@ -82,166 +88,39 @@
-            "The 3D to 1D mapping function divides all 3D space into discrete blocks, and orders these using a space filling curve designed " +
-            "to optimize the requirement that blocks that are close together in this ordered list are also close together in 3D space. " +
-            "This requires that the extents of the 3D space be known in advance and never changed. If you do change these settings, you " +
-            "need to recreate any affected index in order for the settings to apply, otherwise the index will retain the previous settings." )
-    @Internal
-    public static final Setting<Integer> spatial_crs_WGS_84_3D_z_max = setting( "unsupported.dbms.db.spatial.crs.WGS-84-3D.z.max", INTEGER, "1000000" );
-
+    public List<ConfigOptions> getConfigOptions()
+    {
+        String prefix = "unsupported.dbms.db.spatial.crs";
+        ArrayList<ConfigOptions> crsSettings = (ArrayList) LoadableConfig.super.getConfigOptions();
+        double defaultCartesianExtent = 1000000;
+        double[] defaultGeographicExtents = new double[]{180, 90, defaultCartesianExtent};
+        for ( CoordinateReferenceSystem crs : CoordinateReferenceSystem.all() )
+        {
+            char[] coords = ((crs.getDimension() == 2) ? "xy" : "xyz").toCharArray();
+            for ( int dim = 0; dim < coords.length; dim++ )
+            {
+                for ( String rangeName : new String[]{"minimum", "maximum"} )
+                {
+                    String rangeKey = rangeName.substring( 0, 3 );
+                    String descriptionHeader =
+                            String.format( "The %s %s value for the index extents for %dD %s spatial index", rangeName, coords[dim], crs.getDimension(),
+                                    crs.getName().replace( "_3D", "" ) );
+                    String descriptionBody =
+                            "The 2D to 1D mapping function divides all 2D space into discrete tiles, and orders these using a space filling curve designed " +
+                            "to optimize the requirement that tiles that are close together in this ordered list are also close together in 2D space. " +
+                            "This requires that the extents of the 2D space be known in advance and never changed. If you do change these settings, you " +
+                            "need to recreate any affected index in order for the settings to apply, otherwise the index will retain the previous settings.";
+                    String[] keyFields = new String[]{prefix, crs.getName().toLowerCase(), String.valueOf( coords[dim] ), rangeKey};
+                    double defValue = crs.isGeographic() ? defaultGeographicExtents[dim] : defaultCartesianExtent;
+                    defValue = rangeKey.equals( "min" ) ? -1 * defValue : defValue;
+                    Setting setting = setting( String.join( ".", keyFields ), DOUBLE, String.valueOf( defValue ) );
```

Comment:
Why have you changed the type from `INTEGER` to `DOUBLE`?

Reply:
It was always supposed to be DOUBLE. For cartesian we were using integer values by convention but internally they always end up as double in Envelope.